pkg/querier/queryrange/limits.go

```diff
@@ -64,10 +66,15 @@ func WithDefaultLimits(l Limits, conf queryrangebase.Config) Limits {
 		overrides: true,
 	}
 
+	// TODO(ssncferreira): Remove once cortex' split_queries_by_interval is fully deprecated in the next major release
 	if conf.SplitQueriesByInterval != 0 {
-		res.splitDuration = conf.SplitQueriesByInterval
+		level.Warn(util_log.Logger).Log("deprecated", "yaml flag 'query_range.split_queries_by_interval' is deprecated, use yaml flag 'limits_config.split_queries_by_interval' or CLI flag -querier.split-queries-by-interval instead.",
+			"default split_queries_by_interval", l.QuerySplitDurationDefault())
```

Comment:
For users currently using the config file, setting `res.splitDuration = conf.SplitQueriesByInterval` has no effect here. Since limits now contain a default value (30 minutes), this will always be returned, meaning clause: https://github.com/grafana/loki/blob/87cbe50b59f8e85ca979d858b6a4b02ca5b9cbe5/pkg/querier/queryrange/limits.go#L54-L56
will never be reached.

Therefore, for users currently using the config file (while the flag is not removed), I see two options:
1) global default value of 30 minutes is **overwritten** by the value of `conf.SplitQueriesByInterval`
2) Solution here implemented: simply log a warning and use the default value of 30 minutes

What do you think?